kafka/src/test/groovy/io/micronaut/configuration/kafka/annotation/KafkaClientSpec.groovy

```diff
@@ -36,7 +38,7 @@ class KafkaClientSpec extends Specification {
         MyClient client = ctx.getBean(MyClient)
 
         when:
-        client.sendSync("test", "hello-world")
+        client.sendSync("test", "hello-world", [new RecordHeader("hello", "world".bytes)])
```

Comment:
@brianwyka seems silly but what does the test for `[new RecordHeader("hello", "world".bytes), null, new RecordHeader("", null), new RecordHeader(null, null)]` produce?

Reply:
@ctoestreich you can add `, null, new RecordHeader("", null)`, and test still passes, however, `new RecordHeader(null, null)` is not valid since `RecordHeader` does not allow `null` keys.